airflow/jobs.py

```diff
@@ -802,7 +802,8 @@ def create_dag_run(self, dag, session=None):
                     dag.handle_callback(dr, success=False, reason='dagrun_timeout',
                                         session=session)
                     timedout_runs += 1
-            session.commit()
+                    session.merge(dr)
+                    session.commit()
```

Comment:
There is precedent for doing a commit [for each for-loop iteration](https://github.com/github/incubator-airflow/blob/3f2729cde29c6efcace0fc6bba878cae7d72df4d/airflow/jobs.py#L1789-L1790) but I was wondering if you could do one commit after the loop since everything would be merged? I'm sure your code will work 👍

Reply:
Sadly that approach only commits the latest failed DagRun state:

<img width="1017" alt="Screen Shot 2020-02-05 at 8 09 33 AM" src="https://user-images.githubusercontent.com/11540582/73819039-ebad1980-47ee-11ea-8825-d682e748d726.png">